src/core/parser.js

```diff
@@ -1172,55 +1173,58 @@ class Lexer {
   }
 }
 
-var Linearization = {
-  create: function LinearizationCreate(stream) {
-    function getInt(name, allowZeroValue) {
-      var obj = linDict.get(name);
+class Linearization {
+  static create(stream) {
+    function getInt(linDict, name, allowZeroValue = false) {
+      const obj = linDict.get(name);
       if (Number.isInteger(obj) && (allowZeroValue ? obj >= 0 : obj > 0)) {
         return obj;
       }
-      throw new Error('The "' + name + '" parameter in the linearization ' +
+      throw new Error(`The "${name}" parameter in the linearization ` +
                       'dictionary is invalid.');
     }
-    function getHints() {
-      var hints = linDict.get('H'), hintsLength, item;
-      if (Array.isArray(hints) &&
-          ((hintsLength = hints.length) === 2 || hintsLength === 4)) {
-        for (var index = 0; index < hintsLength; index++) {
-          if (!(Number.isInteger(item = hints[index]) && item > 0)) {
-            throw new Error('Hint (' + index +
-                            ') in the linearization dictionary is invalid.');
+
+    function getHints(linDict) {
+      const hints = linDict.get('H'), hintsLength = hints.length;
```

Comment:
Drive-by-comment: Please don't move `hintsLength = hints.length;` here, since that will result in errors being thrown if `hints` isn't actually an Array (which is why the old code checked that *first*).

Reply:
Whoops, good point! I'll adjust that.

Reply:
Fixed in the new commit series.